ably/options.go

```diff
@@ -186,10 +186,13 @@ func (opts *AuthOptions) KeySecret() string {
 	return ""
 }
 
+
 type ClientOptions struct {
 	AuthOptions
 
 	RestHost                string // optional; overwrite endpoint hostname for REST client
+
+	// Deprecated: FallbackHostsUseDefault will be removed in the future
 	FallbackHostsUseDefault bool
```

Comment:
Doc comments in Go [should start with the name being documented](https://golang.org/doc/effective_go.html#commentary):

> 
>  Inside a package, any comment immediately preceding a top-level declaration serves as a doc comment for that declaration. Every exported (capitalized) name in a program should have a doc comment.
> 
> Doc comments work best as complete sentences, which allow a wide variety of automated presentations. The first sentence should be a one-sentence summary that starts with the name being declared.
> 
> ```go
> // Compile parses a regular expression and returns, if successful,
> // a Regexp that can be used to match against text.
> func Compile(str string) (*Regexp, error) {
> ```

Also, I'd suggest pointing users to the non-deprecated alternative.

Reply:
Actually, I referred to the https://github.com/ably/ably-go/blob/681c5a8240349e7b4fe6ea4e7318125383786326/ably/options.go#L233-L234 in clientOptions for deprecation comment

Reply:
Also, there is no alternative specified to the FallbackHostsUseDefault as per spec. FallbackHostsUseDefault option will be removed in the future as the comment says.

Reply:
Ah true, this is the internal `clientOptions` struct, not the public-facing option.

So, how about putting the deprecation notice in the public `WithFallbackHostsUseDefault` function too, where users and tooling may see it?

> Also deprecation comments won't work if they start with property name, you can check
https://rakyll.org/deprecated/

That very article shows how you can do both:

```go
// Execer is an optional interface that may be implemented by a Conn.
//
// [...]
//
// Deprecated: Drivers should implement ExecerContext instead (or additionally).
type Execer interface {
```

> Also, there is no alternative specified to the FallbackHostsUseDefault as per spec

How about something like:

> Deprecated: The library will automatically use default fallback hosts when a custom REST host or custom fallback hosts aren't provided.